elements/lisk-codec/src/codec.ts

```diff
@@ -113,6 +120,42 @@ export class Codec {
 		return (res as unknown) as T;
 	}
 
+	// For performance applications use decode() instead!
+	public decodeJSON = (schema: Schema, message: Buffer): GenericObject => {
+		const decodedMessage: IteratableGenericObject = this.decode(
+			schema,
+			message,
+		);
+		const decodedMessageCopy = objectUtils.cloneDeep(decodedMessage);
+		decodedMessageCopy[Symbol.iterator] = iterator;
+
+		recursiveTypeCast(
+			'toJSON',
+			decodedMessageCopy,
+			(schema as unknown) as SchemaProps,
+			[],
+		);
+		return decodedMessageCopy as GenericObject;
+	};
+
+	// For performance applications use encode() instead!
+	public encodeJSON = (
```

Comment:
```
public encode(schema: Schema, message: object): Buffer
```

Reply:
I think you meant the type for `message` which I changed. If it was not that please comment again :)